src/subplots.ts

```diff
@@ -2045,3 +2075,78 @@ export class Histogram extends Frame {
     return [mouse3X, mouse3Y];
   }
 }
+
+
+export class newScatter extends Frame {
+  public points: newPoint2D[] = [];
+  public tooltipAttr: string[];
+  readonly pointsColorFill: string = 'hsl(203, 90%, 85%)';
+  readonly pointsColorStroke: string = 'hsl(0, 0%, 0%)';
+  constructor(
+    public data: any,
+    public width: number,
+    public height: number,
+    public buttons_ON: boolean,
+    public X: number,
+    public Y: number,
+    public canvas_id: string,
+    public is_in_multiplot: boolean = false
+    ) {
+      super(data, width, height, buttons_ON, X, Y, canvas_id, is_in_multiplot);
+      if (!data.tooltip) {this.tooltipAttr = Array.from(this.features.keys()) }
+      else this.tooltipAttr = data.tooltip.attribute;
+    }
+
+  private computePoints(): newPoint2D[] {
+    const numericVectorX = this.axes[0].stringsToValues(this.features.get(this.xFeature));
+    const numericVectorY = this.axes[1].stringsToValues(this.features.get(this.yFeature));
+    const points: newPoint2D[] = [];
+    const pointsHTMatrix = this.getPointsHTMatrix(numericVectorX, numericVectorY);
+    const coordsMatrix = this.projectPoints(pointsHTMatrix);
+    for (let index = 0 ; index < numericVectorX.length ; index++) {
+      let newPoint = new newPoint2D(coordsMatrix.get(0, index),  coordsMatrix.get(1, index), 5, "circle", undefined, "hsl(203, 90%, 85%)");
+      if (this.hoveredIndices[index]) {newPoint.isHovered = true};
+      if (this.clickedIndices[index]) {newPoint.isClicked = true};
+      if (this.selectedIndices[index]) {newPoint.isSelected = true};
+
+      const inCanvasX = numericVectorX[index] < this.axes[0].maxValue && numericVectorX[index] > this.axes[0].minValue;
+      const inCanvasY = numericVectorY[index] < this.axes[1].maxValue && numericVectorY[index] > this.axes[1].minValue;
+      if (inCanvasX && inCanvasY) {
+        if (newPoint.isClicked) this.tooltipAttr.forEach(attr => newPoint.tooltipMap.set(attr, this.features.get(attr)[index]));
+        newPoint.draw(this.context_show);
+      }
+      points.push(newPoint);
+    }
+    return points
+  }
+
+  private getPointsHTMatrix(vectorX: number[], vectorY: number[]) {
+    return new matrices.Matrix([vectorX, vectorY, Array.from(Array(this.nSamples), () => 1)])
+  }
+
+  private projectPoints(pointsMatrix: matrices.Matrix): matrices.Matrix {
+    const relativeMatrix = new matrices.Matrix([
+      [this.relativeMatrix.a, this.relativeMatrix.c, this.relativeMatrix.e],
+      [this.relativeMatrix.b, this.relativeMatrix.d, this.relativeMatrix.f],
+      [0, 0, 1]
+    ])
+    return relativeMatrix.mmul(pointsMatrix)
+  }
+
+  public stateUpdate(context: CanvasRenderingContext2D, objects: any[], mouseCoords: Vertex, stateName: string, keepState: boolean, invertState: boolean): void {
+    // TODO: Fast and dirty implementation. Needs to be rethought ?
```

Comment:
I actually am struggling to keep track of what is going on here

Reply:
It has been removed with the above modifications